### DISCUSSION TITLE
Using predefined node shapes

### DISCUSSION THREAD

**Discussion by ynnckth** [xyflow/xyflow#2608] [General]
According to the [docs](https://reactflow.dev/docs/examples/nodes/shapes/), I understand it should be possible to specify one of the predefined node shapes for circle, round-rect, hexagon, diamond and a few others. I understand this should be possible by setting the 'shape' property in the node data. However, this doesn't seem to have any effect. 

I'm defining my nodes as follows and would expect the first node to be rendered as a circle, but it's rendered as a standard rectangle: 

```javascript
import { useCallback } from "react";
import ReactFlow, {
  MiniMap,
  Controls,
  Background,
  useNodesState,
  useEdgesState,
  addEdge
} from "reactflow";
import "reactflow/dist/style.css";

const initialNodes = [
  {
    id: "1",
    position: { x: 0, y: 0 },
    data: { label: "circle", shape: "circle" }
  },
  { id: "2", position: { x: 0, y: 100 }, data: { label: "2" } }
];

const initialEdges = [{ id: "e1-2", source: "1", target: "2" }];

export default function App() {
  const [nodes, setNodes, onNodesChange] = useNodesState(initialNodes);
  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);

  const onConnect = useCallback(
    (params) => setEdges((eds) => addEdge(params, eds)),
    [setEdges]
  );

  return (
    <div className="App" style={{ width: "100vw", height: "100vh" }}>
      <ReactFlow
        nodes={nodes}
        edges={edges}
        onNodesChange={onNodesChange}
        onEdgesChange={onEdgesChange}
        onConnect={onConnect}
      >
        <MiniMap />
        <Controls />
        <Background />
      </ReactFlow>
    </div>
  );
}
```

Thanks for any hints!

**Comment by amintabrizi** [comment]
You have to do some additional steps to define circle node.
There are good examples for creating custom nodes, Look these two links:
https://reactflow.dev/docs/guides/custom-nodes/
https://reactflow.dev/docs/examples/nodes/custom-node/ 
Also, I have created a circle node in the below link:
https://codesandbox.io/s/reactflow-circle-node-yqzrxi
 I will briefly tell you the steps to create a dedicated node, but it is better that you take a closer look at the documents.
1- Creating a custom circle component
In the example above, file 'CircleNode.js' refers to the same issue. I have created the circle by CSS, but you can do it with SVG or Canvas too.
2- Create 'nodeTypes' outside the component
Pay attention to lines 16 and 82 in the 'app.js' 
3-Finally, you have to use the new type in the list of nodes
Refer to line 35 in the 'app.js'
